components/compositions/layer-parser.service.js

```diff
@@ -83,6 +92,42 @@ export default function (HsMapService, HsAddLayersVectorService) {
       HsMapService.proxifyLayerLoader(new_layer, !lyr_def.singleTile);
       return new_layer;
     },
+    /**
+     * @ngdoc method
+     * @name hs.compositions.config_parsers.service#createWMTSLayer
+     * @public
+     * @param {object} lyr_def Layer definition object
+     * @returns {object} Ol Tile layer
+     * @description Parse definition object to create WMTS Ol.layer  (source = ol.source.WMTS)
+     */
+    createWMTSLayer: async function (lyr_def) {
+      const wmts = new Tile({
+        title: lyr_def.title,
+        info_format: lyr_def.info_format,
+        source: new WMTS({}),
+      });
+
+      // Get WMTS Capabilities and create WMTS source base on it
+      const source = await HsWmtsGetCapabilitiesService.requestGetCapabilities(
+        lyr_def.url
+      ).then((res) => {
+        //parse the XML response and create options object...
+        const parser = new WMTSCapabilities();
+        const result = parser.read(res.data);
+        // ...create WMTS Capabilities based on the parsed options
+        const options = optionsFromCapabilities(result, {
+          layer: lyr_def.layer,
+          matrixSet: lyr_def.matrixSet,
+          format: lyr_def.format,
+        });
+        // WMTS source for raster tiles layer
+        return new WMTS(options);
+      });
```

Comment:
Quite complicated combination of `then` and `await` to my taste..

Reply:
Yea a guess so. Made some changes..feels a bit less complicated now https://github.com/hslayers/hslayers-ng/pull/1393/commits/1ff149b40e612d5c765d06ad0b410cdd9d267238. ...aslo was able to get rid of all awaits for jsonToLayer which I wasnt real fan of